### ISSUE TITLE
CMA on computationally expensive functions

### ISSUE THREAD

**Issue by JacobLoe** [CMA-ES/pycma#192]
I have a question on how pycma uses the objective function internally. 
My function has an input dimension of 120 parameters, but I'm not even close to being able to optimize the whole function. 

If I write a wrapper around my function so CMA only has to optimize a subset of the original, CMA works fine with 15 parameters. 
At 30 my laptop grinds to a halt and is borderline unuseable. With 120 the process is killed before even really starting.

A quadratic function with 120 parameters works perfectly fine, so the problem cannot be the number of parameters itself I assume, rather I assume CMA somehow duplicates my function and hogs up my RAM/CPU. 
Is there a way around this ? Or do I have to focus on making my function more lightweight ?

Any help/comments are appreciated.

**Comment by nikohansen** [comment]
Pasting the calling code here, how the `cma` module was used, would be helpful for diagnostic purposes. The `cma` module doesn't really do anything with the function, so I suspect the reason lies within the function.

You may consider using the lighter `.optimize` call as interface, or even the `ask` and `tell` interface such that the function call is not hidden at all, see [here](https://github.com/CMA-ES/pycma/blob/master/cma/evolution_strategy.py#L1111) for an example of both.

**Comment by JacobLoe** [comment]
Thanks for the help, I'll look into your suggestion.

This is the only time `cma` is called. I'm not sure if the problem is that I'm using `fmin`.
`    ea = cma.fmin(objective_function, initial_population, sigma0=sigma,
                  options={'maxiter': max_iterations, 'verbose': -8, 'bounds': bounds},
                  args=(n_simulations, node_indices, n_nodes, eval_function, statistic, total_budget))`

**Comment by JacobLoe** [comment]
I'm now using a more powerful machine which can handle my function.
Between changing machines I haven't changed how I used ```cma```, so I still have no clue where the problem was.
I then switched to use the ```ask``` and ```tell``` interface and now my code also works on the first machine, which is surprising to me because I assumed those are essentially the same, just that ```ask/tell``` is more granular.
